dropwizard-app/src/test/java/org/coner/resource/HandicapGroupsResourceTest.java

```diff
@@ -0,0 +1,132 @@
+import io.dropwizard.testing.FixtureHelpers;
+import io.dropwizard.testing.junit.ResourceTestRule;
+import org.coner.api.response.ErrorsResponse;
+import org.coner.boundary.HandicapGroupBoundary;
+import org.coner.core.ConerCoreService;
+import org.coner.core.domain.HandicapGroup;
+import org.coner.util.JacksonUtil;
+import org.eclipse.jetty.http.HttpStatus;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.math.BigDecimal;
+
+import static org.fest.assertions.Assertions.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ *
+ */
+public class HandicapGroupsResourceTest {
+    private final HandicapGroupBoundary handicapGroupBoundary = mock(HandicapGroupBoundary.class);
+    private final ConerCoreService conerCoreService = mock(ConerCoreService.class);
+
+    private ObjectMapper objectMapper;
+
+    @Rule
+    public final ResourceTestRule resources = ResourceTestRule.builder()
+            .addResource(new HandicapGroupsResource(handicapGroupBoundary, conerCoreService))
+            .addProvider(new ConstraintViolationExceptionMapper())
+            .build();
+
+    @Before
+    public void setup() {
+        reset(handicapGroupBoundary, conerCoreService);
+
+        objectMapper = Jackson.newObjectMapper();
+        JacksonUtil.configureObjectMapper(objectMapper);
+    }
+
+    @Test
+    public void whenAddValidHandicapGroupItShouldAddHandicapGroup() throws Exception {
+        org.coner.api.entity.HandicapGroup requestHandicapGroup = objectMapper.readValue(
+                FixtureHelpers.fixture("fixtures/api/entity/handicap_group_add-request.json"),
+                org.coner.api.entity.HandicapGroup.class
+        );
+
+        Entity<org.coner.api.entity.HandicapGroup> requestEntity = Entity.json(requestHandicapGroup);
+
+        org.coner.api.entity.HandicapGroup responseHandicapGroup = objectMapper.readValue(
+                FixtureHelpers.fixture("fixtures/api/entity/handicap_group_add-response.json"),
+                org.coner.api.entity.HandicapGroup.class
+        );
+        org.coner.core.domain.HandicapGroup requestHandicapGroupAsDomain = new org.coner.core.domain.HandicapGroup();
+        requestHandicapGroupAsDomain.setId("arbitrary-id-from-service");
+        requestHandicapGroupAsDomain.setName("C Street");
+        requestHandicapGroupAsDomain.setHandicapFactor(BigDecimal.valueOf(0.75));
+
+        when(handicapGroupBoundary.toDomainEntity(requestHandicapGroup)).thenReturn(requestHandicapGroupAsDomain);
+        when(handicapGroupBoundary.toApiEntity(any(org.coner.core.domain.HandicapGroup.class)))
+                .thenReturn(responseHandicapGroup);
```

Comment:
This seems to be the only other reference to `responseHandicapGroup`. The mocked scenario doesn't match anything in the implementation so it should probably go away